polars/polars-time/src/windows/groupby.rs

```diff
@@ -355,7 +355,7 @@ pub(crate) fn groupby_values_iter_partial_lookbehind<'a>(
         let b = Bounds::new(lower, upper);
 
         for &t in &time[lagging_offset..] {
-            if b.is_member(t, closed_window) || lagging_offset == i {
+            if b.is_member(t, closed_window) || (offset.negative && lagging_offset == i) {
```

Comment:
Can we branch on `offset.negative` outside the loop?

Then we have two loops that compute the `lagging_offset`. The loops are super small, so almost no code bloat.

Reply:
Can we actually get here if `offset.negative` isn't set? :thinking:

Reply:
only in this test

https://github.com/pola-rs/polars/blob/dc7bfd2ec362ec6fca7ebe0ff470cc11ee16f66c/polars/polars-time/src/windows/test.rs#L711-L734

maybe it's better to just remove that test then?

Reply:
I've removed this part of the test, reverted the change to `groupby_values_iter_partial_lookbehind`, and expanded the tests on the python side by doing quite a bit more parametrisation